### ISSUE TITLE
getByRole doesn't work for menu with aria-label

### ISSUE THREAD

**Issue by sandrina-p** [testing-library/dom-testing-library#1078]
- `@testing-library/dom` version: 4.1.0
- Testing Framework and version: Jest (The same as in the codesandbox below)

### Relevant code or config:

```js
    <ul role="menu" aria-label="actions">
      <li role="menuitem">copy</li>
      <li role="menuitem">paste</li>
      <li role="menuitem">cut</li>
    </ul>
    
    <ul role="menu" aria-label="share">
      <li role="menuitem">copy link</li>
      <li role="menuitem">report bug</li>
    </ul>
```

### What you did:

```
screen.getByRole('menu', { name: 'actions' }) 
```
### What happened:

The element was not found. 
> Unable to find an accessible element with the role "menu" and name "actions"

I was expecting to be found because, according to the Chrome's Accessibility Tree, the role and name are set correctly: 

![image](https://user-images.githubusercontent.com/14869087/145786006-956d6cf1-57fc-4e10-8a6e-1603d9e76007.png)

### Reproduction:

- Condesabox: https://codesandbox.io/s/react-testing-library-menu-name-6ir07
- Testing Playground: https://testing-playground.com/gist/ba36265b023b1e0bb5dbe513cceb84f1/d07dc90240b88c2c71ad7c1478a7e40ba800778d

### References:
This was originally discovered at [Discord KCD](https://discord.com/channels/715220730605731931/821796009880453120/919872873072566302) by Thomas.

**Comment by eps1lon** [comment]
Thanks for the report

Does this also reproduce locally? It can't even find the `menu` role which is probably a duplicate of https://github.com/testing-library/dom-testing-library/issues/1032 i.e. an issue with the testing environment of Codesandbox (https://github.com/codesandbox/codesandbox-client/issues/5887).

**Comment by MatanBobi** [comment]
@eps1lon I checked now and this passed in local env for me.

**Comment by thomasmery** [comment]
hi there,

I am trying to reproduce this locally and my test fails whenever I try to access a menu by role & name
```jsx

import React from 'react';
import { render, screen } from '@testing-library/react';

const Component = () => (
  <>
    <ul role="menu" aria-label="actions">
      <li role="menuitem">copy</li>
      <li role="menuitem">paste</li>
      <li role="menuitem">cut</li>
    </ul>

    <ul role="menu" aria-label="share">
      <li role="menuitem">copy link</li>
      <li role="menuitem">report bug</li>
    </ul>
  </>
);

test('getByRoleMenu', () => {
  render(<Component />);
  expect(screen.getByRole('menu', { name: 'actions' })).toBeInTheDocument();
  expect(screen.getByRole('menu', { name: 'share' })).toBeInTheDocument();
});

```

![image](https://user-images.githubusercontent.com/1503822/145799200-b1bddd24-f5e1-420a-a3e7-1d26ba46d6b9.png)

am I doing something wrong?

@MatanBobi would you share the code of your test?

thanks

**Comment by eps1lon** [comment]
@thomasmery Could you post the result of `npx testing-library-envinfo`? You might have an older version of `dom-accessibility-api` that fixed https://github.com/testing-library/dom-testing-library/issues/1059

**Comment by thomasmery** [comment]
here it is 

```shell
System:
    OS: macOS 11.6
  Binaries:
    Node: 14.1.0 - ~/.nvm/versions/node/v14.1.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.1.0/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v14.1.0/bin/npm
  npmPackages:
    @testing-library/cypress: ^7.0.6 => 7.0.6 
    @testing-library/dom: ^7.21.4 => 7.31.2 
    @testing-library/jest-dom: ^5.11.10 => 5.13.0 
    @testing-library/react: ^11.2.6 => 11.2.7 
    @testing-library/user-event: ^12.0.11 => 12.8.3 
    aria-query:  4.2.2 
    dom-accessibility-api:  0.5.6 
    jest:  26.6.0 
    jsdom:  16.6.0 
    react: ^17.0.1 => 17.0.2 
    react-dom: ^17.0.1 => 17.0.2 
```

**Comment by MatanBobi** [comment]
> 

@thomasmery The fix for `role="menu"` as @eps1lon mentioned was introduced in [dom-accessibility-api version 0.5.9](https://github.com/eps1lon/dom-accessibility-api/releases/tag/v0.5.9). You'll need to upgrade `@testing-library/dom` to version `^8.10.1` to get it :)

**Comment by thomasmery** [comment]
thanks!

test is passing now

I had to upgrade to: 

 ```bash
    @testing-library/dom: ^8.11.1 => 8.11.1 
    @testing-library/jest-dom: ^5.16.1 => 5.16.1 
    @testing-library/react: ^12.1.2 => 12.1.2
    dom-accessibility-api: ^0.5.10 => 0.5.10 ``
```
